vipe/oozie/reader_handlers.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2013-2015 University of Warsaw
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+__author__ = "Mateusz Kobos mkobos@icm.edu.pl"
+
+from vipe.oozie.parse_utils import properties_to_dict
+from vipe.oozie.graph import DecisionCase, Decision, End, Kill, Fork, Join, \
+    Start, OtherAction, SubworkflowAction, JavaAction, StreamingMapReduceAction, \
+    JavaMapReduceAction, PigAction, HiveAction, FSAction, DistCPAction
+
+def handle_action(elem):
+    """
+    Returns:
+        a (name, Action) pair where name is a string.
+    """
+    name = elem.get('name')
+    type_ = __get_action_type(elem)
+    ok_node = elem.find('ok').get('to')
+    error_node = elem.find('error').get('to')
+    action_elem = elem.find(type_)
+    configuration = __get_configuration(action_elem)
+    if type_ == 'sub-workflow':
+        app_path = action_elem.find('app-path').text
+        return (name, SubworkflowAction(ok_node, error_node, configuration, 
+                                 app_path))
+    elif type_ == 'java':
+        main_class = action_elem.find('main-class').text
+        args = [e.text for e in action_elem.findall('arg')]
+        captures_output = action_elem.find('capture-output') is not None
+        return (name, JavaAction(ok_node, error_node, configuration, 
```

Comment:
I don't know yet. I assumed that this is somewhat exceptional case when the 'capture-output' tag is present (although it's a standard thing for schema ganerator nodes) so it might make sense to process such node in a special way.